src/ajax.js

```diff
@@ -270,7 +270,23 @@ function ajaxConvert( s, response, jqXHR, isSuccess ) {
 		}
 	}
 
-	return { state: "success", data: response };
+	if ( jQuery.isThenable(response) ) {
+		return response.then(
+			function(response) {
+				this.responseJSON = response;
+				return { state: "success", data: response };
+			}.bind(jqXHR),
+			function(e) {
+				if ( this[ "throws" ] ) {
+					throw e;
```

Comment:
This doesn't smell right... assuming a valid `.then` implementation, rethrowing is a no-op and very likely invisible anyway.